### PR TITLE
Mention end to end tests in docs

### PR DESCRIPTION
**Pull request by bocekm**
.. to complete the picture of what testing we have for Leapp.

### CODE REVIEW

**Comment by AloisMahdal** [comment]
The content of the mention is correct, although I'm not sure about the reason why mention it in the first place?  Assuming the audience is general upstream community, wouldn't adding another category that is not very useful to them make things confusing?

**Comment by bocekm** [comment]
The audience is also internal Red Hat community (and I assume it will be the majority) as we point to this document from many other places. Without this change, the information about the tests would be simply incomplete.

**Comment by AloisMahdal** [comment]
I'm not sure why this should be reviewed by OAMG-QE.  Sure, the title seems that the change is relevant to us, but then the actual commit overreaches it by rewriting the whole page. So 90% of the commit is about unit/component tests which are completely outside our range.

**Comment by bocekm** [comment]
Ok, I've added the other reviewers as well.

File: docs/tests.md [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
 
-## Component tests
-   - These tests provide fabricated input messages for the actor, check the outputs stated in the actor's description.
-   - These tests should not be written based on the actor's code but rather based on the behavior stated in the actor's description. Ideally they could be written by somebody who doesn't know the code.
-   - Example of a component test: https://github.com/oamg/leapp-repository/blob/master/repos/system_upgrade/el7toel8/actors/redhatsignedrpmcheck/tests/test_redhatsignedrpmcheck.py
-
-## Unit tests
+## Unit and component tests
+- Both unit and component tests are to be placed in the actor's _tests_ folder.
+- Tutorial on how to write unit and component tests: https://leapp.readthedocs.io/en/latest/unit-testing.html
```
Comment (artmello) [inline docs/tests.md]: Could we rename the "unit-testing" part of the URL? This is sort of misleading with current content.

Comment (bocekm) [reply docs/tests.md]: Yes, I agree. I thought about that. But I meant to change it on all the places in one push. In separate PRs.

File: docs/tests.md [hunk 2]
```diff
@@ -1,17 +1,22 @@
 # Tests for actors
 
-You can implement two types of tests for an actor in the leapp-repository - component and unit tests.
+The Leapp actors are currently covered by three types of tests - unit, component and e2e.
```
[the remark this answers is not in the document]
Comment (bocekm) [reply docs/tests.md]: Makes sense.

File: docs/tests.md [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
+## Unit and component tests
+- Both unit and component tests are to be placed in the actor's _tests_ folder.
+- Tutorial on how to write unit and component tests: https://leapp.readthedocs.io/en/latest/unit-testing.html
+   
+### Unit tests
    - These tests deal with individual actor's functions/methods.
-   - You can't unit test any method/function within the actor.py. You can write unit tests only for the functions/methods within actor's libraries.
-   - Thus, to be able to write unit tests for an actor, ideally the only thing in the actor.py's _process()_ method is calling the entry-point function of the actor's library python module.
-   - Tutorial on how to write unit tests: https://leapp.readthedocs.io/en/latest/unit-testing.html
+   - Currently it's not possible to unit test any method/function within the *actor.py*. You can write unit tests only for functions/methods within the actor's libraries.
```
[the remark this answers is not in the document]
Comment (bocekm) [reply docs/tests.md]: Well, it's supposed to change in near future. @vinzenz has some plans to change that. There's already a first step in that direction: https://github.com/oamg/leapp/pull/408.
But I can remove the "currently". You're right that it's not necessary to tease anyone.

[the remark this answers is not in the document]
Comment (bocekm) [reply docs/tests.md]: Removed.

File: docs/tests.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (bocekm) [reply docs/tests.md]: Fixed.

**Pull request merged**